src/basics.ml

```diff
@@ -143,16 +146,21 @@ let iter_ctxt : (tvar list -> term -> unit) -> term -> unit = fun action t ->
     | Vari(_)
     | Type
     | Kind
-    | Symb(_)    -> ()
+    | Symb(_)     -> ()
     | Patt(_,_,ts)
     | TEnv(_,ts)
-    | Meta(_,ts) -> Array.iter (iter xs) ts
+    | Meta(_,ts)  -> Array.iter (iter xs) ts
     | Prod(a,b)
-    | Abst(a,b)  ->
+    | Abst(a,b)   ->
        iter xs a;
        let (x,b') = Bindlib.unbind b in
        iter (if Bindlib.binder_occur b then x::xs else xs) b'
-    | Appl(t,u)  -> iter xs t; iter xs u
+    | Appl(t,u)   -> iter xs t; iter xs u
+    | LLet(t,a,u) ->
+        iter xs a;
+        iter xs t;
```

Comment:
permute this 2 lines so that iter* functions always start with t and then a like in the following functions